### PR TITLE
Add squeeze v15 op spec

### PR DESCRIPTION
**Pull request by barnasm1**
### Details:
 - Add squeeze v15 op spec

### Tickets:
 - [*154021*](https://jira.devtools.intel.com/browse/CVS-154021)

### CODE REVIEW

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    Behavior before 2024.3 OpenVINO release: Error is raised when dimension to squeeze is not compatible with 1.
+
+.. note::
+
+    - If index of the dimension to squeeze is provided as a constant input and it points to a dynamic dimension that might be `1`, then the dimension is considered as squeezable. Therefore the rank of the output shape will be reduced, but not dynamic.
+    - If the input with indices is empty or not provided, dynamic dimension compatible with `1` leads to dynamic rank of the output shape.
+
+
+**Attributes**:
+* *allow_axis_skip*
```
Comment (mmikolajcz) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:32]: This line is not properly formatted in built docs

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  * **Description**: Shape inference result dynamic rank if selected axis has value 1 in range of its dynamic.
+  * **Range of values**: ``false`` or ``true``
+  * **Type**: ``boolean``
+  * **Required**: *no*
```
Comment (mmikolajcz) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:37]: Please provide default value for optional attributes

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+*   **1**: Tensor with squeezed values of type *T*.
+
+**Types**
+
+* *T*: any numeric type.
+
+* *T_INT*: any supported integer type.
+
+**Example**
```
Comment (mmikolajcz) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:55]: Please add attribute value to example IR. Consider adding additional examples that will show differences between attribute values and maybe examples of dynamic rank or ignored non-1 dimensions

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+**Detailed description**: *Squeeze* can be used with or without the second input tensor.
+
+* If only the first input is provided, every dimension that is equal to 1 will be removed from it.
+* With the second input provided, each value is an index of a dimension from the first tensor that is to be removed. Specified dimension should be equal to 1, otherwise it will be ignored and copied as is.
+  Dimension indices can be specified directly, or by negative indices (counting dimensions from the end).
+
+.. note::
+
+    Behavior before 2024.3 OpenVINO release: Error is raised when dimension to squeeze is not compatible with 1.
```
Comment (mmikolajcz) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:23]: Is this mentioned version of OV correct? If I understand correctly 24.4 does not include updated v15::Squeeze op

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 1 above)
Comment (barnasm1) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:33]: Done

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 4 above)
Comment (barnasm1) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:23]: Done

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 3 above)
Comment (barnasm1) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:53]: Done

**Review by mmikolajcz — APPROVED** [review]

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+    - If index of the dimension to squeeze is provided as a constant input and it points to a dynamic dimension that might be `1`, then the dimension is considered as squeezable. Therefore the rank of the output shape will be reduced, but not dynamic.
+    - If the input with indices is empty or not provided, dynamic dimension compatible with `1` leads to dynamic rank of the output shape.
+
+
+**Attributes**:
+
+* *allow_axis_skip*
+
+  * **Description**: Shape inference result dynamic rank if selected axis has value 1 in range of its dynamic.
```
Comment (p-wysocki) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:31]: Just a suggestion, it's up to whether to apply it.
```suggestion
  * **Description**: If true, shape inference results in a dynamic rank if selected axis has value 1 in its dynamic range.
```

**Review by p-wysocki — APPROVED** [review]

**Comment by mmikolajcz** [comment]
PR that adds opset15 to documentation was merged, please update following files to link added specification:

- docs/articles_en/documentation/openvino-ir-format/operation-sets/available-opsets/opset15.rst
- docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs.rst

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+.. note::
+
+    - If index of the dimension to squeeze is provided as a constant input and it points to a dynamic dimension that might be `1`, then the dimension is considered as squeezable. Therefore the rank of the output shape will be reduced, but not dynamic.
+    - If the input with indices is empty or not provided, dynamic dimension compatible with `1` leads to dynamic rank of the output shape.
+
+
+**Attributes**:
+
+* *allow_axis_skip*
+
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:30]: Do we expect exception raised when `allow_axis_skip=False` and given axis corresponds to dimension that is not squeezable?

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 7]
```diff
@@ -0,0 +1,174 @@
+Squeeze
+=======
+
```
Comment (rkazants) [inline docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:3]: why do we require such version? Is there similar operation in frameworks like PyTorch, TF, ONNX? Can we refer a link to such operation?

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 6 above)
Comment (mitruska) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:30]: It could be implemented to throw, but the intention is to keep the v0::Squeeze compatible behavior when `allow_axis_skip=False`.
For model enablement the v0::Squeeze has been already updated to be not so strict for static dimensions, but it was not possible to update the all dynamic cases and keep backward compatibility. Especially previous Squeeze logic assumes that dynamic dim with 1 in range like {-1} or {1-2} is squeezable, and produces static rank. 
While the recent request to align with torch is to produce dynamic rank for such case.
In fact `allow_axis_skip=False` can be understood like `assume_dynamic_dim_is_squeezable=True`.

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 7 above)
Comment (mitruska) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:3]: It's needed for alignment with torch, more details can be found in the internal ticket: 143585.

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 5 above)
Comment (barnasm1) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:31]: Done

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+**Detailed description**: *Squeeze* can be used with or without the second input tensor.
+
+* If only the first input is provided, every dimension that is equal to 1 will be removed from it.
+* With the second input provided, each value is an index of a dimension from the first tensor that is to be removed. Specified dimension should be equal to 1, otherwise it will be ignored and copied as is.
+  Dimension indices can be specified directly, or by negative indices (counting dimensions from the end).
+
+.. note::
+
+    - If index of the dimension to squeeze is provided as a constant input and it points to a dynamic dimension that might be `1`, then the dimension is considered as squeezable. Therefore the rank of the output shape will be reduced, but not dynamic.
```
[the remark this answers is not in the document]
Comment (barnasm1) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst]: Done

**Review by mitruska — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

File: docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst (hunk 6 above)
Comment (rkazants) [reply docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/shape/squeeze-15.rst:30]: It looks that this mode is also helpful for TF Squeeze op (https://www.tensorflow.org/api_docs/python/tf/raw_ops/Squeeze) when axis = []. For such case, I can compute axes for our Squeeze using `ShapeOf` and `Range` ops and set this mode `allow_axis_skip=False`. This way it will squeeze all dims that are possible to squeeze. So we can squeeze dimension during both `conversion/loading` and `inference`, right?
